GSoC_19/posts/websocket-passive-scan-using-scripts-with-zap.org

```diff
@@ -0,0 +1,845 @@
+
+    Case 1: Write a email (like mail@zaproxy.com) and send them. See what alert
+    have been raised. 
+
+    Case 2: Encode "Hello World" with Base64 (SGVsbG8gV29ybGQ=) and sent them. 
+    See what alert have been raised. 
+    
+    Case 3: Type some valid XML with the following comment in it : 
+    <!-- TODO: fix this issue -->. See what alert have been raised.
+
+    #+BEGIN_SRC xml
+      <layer1>
+        <!--This is a comment -->
+        <layer2>
+          <!-- TODO: fix this issue -->
+          <form> ........ </form>
+        </layer2>
+      </layer1>
+    #+END_SRC
+
+    Case 4: Add a user in the 'Context'([[https://github.com/zaproxy/zap-core-help/wiki/HelpStartConceptsAuthentication#formBased][wiki]]). Encode Username with MD5 hash and send
+    them. See what alert have been raised. 
+
+    Finally: Go to Alerts Panel select all Alerts, right click on them and hit Delete.
+    
+** Script Console Extension
+
+   And now one of my favorite features of ZAP, the Script Console extension.
+
+   #+BEGIN_QUOTE
+   The Script Console add-on allows you to run scripts that can be embedded within ZAP and can access internal ZAP data structures. It supports any scripting language that supports JSR 223 (http://www.jcp.org/en/jsr/detail?id=223) , including:
+
+   ECMAScript / Javascript (using Nashorn engine, included by default)
+
+   Zest https://developer.mozilla.org/en-US/docs/zest (included by default)
+
+   Groovy http://groovy-lang.org/
+
+   Python http://www.jython.org
+
+   Ruby - http://jruby.org/
+
+   and many more...
+   
+   /from [[https://github.com/zaproxy/zap-core-help/wiki/HelpAddonsScriptsScripts][wiki]]/
+   #+END_QUOTE
+
+   Referring to scripts console extension we are referring to the ability that
+   user has to write scripts which implement various behaviors like scans, fuzzing etc.
+   Scripts Console extension has a sidebar where the scripts are displayed. The sidebar
+   has a tree structure, which has two branches, the =Scripts= branch and
+   =Templates=. In both branches there are some folders which are referring on
+   different kind of scripts. As you can see those folders in both branches have
+   the same names.  
+
+   #+ATTR_HTML: :align center :width 300 height 800
+   #+CAPTION: Scripts Tree Sidebar
+   [[../../images/pscan-tutorial/scripts_tree.png]]
+
+   ZAP defines =Templates= in order to be used as base scripts. As we will see in
+   a while every kind of script should implement a specific functionality, named
+   rule. In addition, =Templates= are not runable, enablable or directly
+   editable, they are used only as templates. In order to use a template, right
+   click on it and then hit =New script...=. 
+
+   **Hands On Section:**
+   Go to the template script =Base63 Disclosure.js=, right click on it, change
+   =Script Name= to =MyBase64 Disclosure.js= and hit save. See what happened.
+
+   =Scripts=, on the other hand, are the actual scripts which are running in
```

Comment:
`which are run by ZAP.`? It depends on the script types, they might be executed in the main thread of the functionality.